src/clj/rems/db/resource.clj

```diff
@@ -1,18 +1,34 @@
 (ns rems.db.resource
   (:require [rems.db.core :as db]
-            [rems.util :refer [getx-user-id]]))
+            [rems.InvalidRequestException]
+            [rems.util :refer [getx-user-id]])
+  (:import (org.postgresql.util PSQLException)
+           (rems InvalidRequestException)))
 
 (defn get-resources [filters]
   (->> (db/get-resources)
        (map db/assoc-active)
        (db/apply-filters filters)))
 
+(defn duplicate-resid? [^Exception e]
+  (let [cause (.getCause e)]
+    (and (instance? PSQLException cause)
+         (= "duplicate key value violates unique constraint \"resource_resid_u\""
+            (.getMessage (.getServerErrorMessage ^PSQLException cause))))))
+
 (defn create-resource! [{:keys [resid organization licenses]}]
-  (let [id (:id (db/create-resource! {:resid resid
-                                      :organization organization
-                                      :owneruserid (getx-user-id)
-                                      :modifieruserid (getx-user-id)}))]
-    (doseq [licid licenses]
-      (db/create-resource-license! {:resid id
-                                    :licid licid}))
-    {:id id}))
+  (try
+    (let [id (:id (db/create-resource! {:resid resid
+                                        :organization organization
+                                        :owneruserid (getx-user-id)
+                                        :modifieruserid (getx-user-id)}))]
+      (doseq [licid licenses]
+        (db/create-resource-license! {:resid id
+                                      :licid licid}))
+      {:success true
+       :id id})
+    (catch Exception e
+      (if (duplicate-resid? e)
```

Comment:
This is ok now, but if we start to look at more error cases a separate error-handler were exceptions were passed would make more sense rather than checking all possible error situations where the error happened?

Reply:
I would worry about that then. It's not possible to see yet what it would look like?